### PR TITLE
clean up db and pkgstore + add platform filter in search

### PR DESCRIPTION
**Pull request by madhur-tandon**

### CODE REVIEW

File: quetz/authorization.py [hunk 1]
```diff
@@ -285,6 +285,9 @@ def assert_set_channel_size_limit(self):
             msg="only server maintainer or owner can set channel size limit",
         )
 
+    def assert_channel_db_cleanup(self, channel_name):
+        self.assert_channel_roles(channel_name, [OWNER, MAINTAINER])
```
Comment (wolfv) [inline quetz/authorization.py:289]: I think only OWNER should be allowed here (maintainer might not be able to see the logs of the serveretc).

File: quetz/pkgstores.py [hunk 2]
```diff
@@ -426,3 +454,15 @@ def get_filemetadata(self, channel: str, src: str):
         etag = infodata['etag']
 
         return (msize, mtime, etag)
+
+    def cleanup_temp_files(self, channel: str):
```
Comment (wolfv) [inline quetz/pkgstores.py:458]: this is a bit code duplication here. is there a way to make this a free function?

**Comment by wolfv** [comment]
I went through all of the code and it looks really nice!

I am wondering if we can move the cleanup functions to their own python file (cleanup.py or something like that?).

Also, we could move the pkgstore cleaning function there.

File: quetz/authorization.py (hunk 1 above)
Comment (madhur-tandon) [reply quetz/authorization.py:289]: Done.

File: quetz/pkgstores.py (hunk 2 above)
Comment (madhur-tandon) [reply quetz/pkgstores.py:458]: we should tackle duplication in `S3` store and `Azure` store in a separate PR since this is a part of that now.

**Pull request merged**